src/repositories/TodoRepository.ts

```diff
@@ -1,27 +1,40 @@
 import { TodoEntity } from "../entities/TodoEntity";
 import { IRepository } from "./IRepository";
-import { IDataStorage } from '../dataStorages/IDataStorage';
+import { DataStorageId, IDataStorage } from '../dataStorages/IDataStorage';
+import { IEntity } from "src/entities/IEntity";
 
 
 export class TodoRepository implements IRepository<TodoEntity>{
 
     constructor(private dataStorage: IDataStorage<TodoEntity>) {}
 
-    async getAll(): Promise<TodoEntity[]> {
+    async getAll(): Promise<Required<TodoEntity[]>> {
         const result = await this.dataStorage.find()
-        console.log(result)
         return result
     }
 
-    async insertOne(newTodo: string): Promise<TodoEntity> {
-        return await this.dataStorage.create(newTodo)
+    async insertOne(newTodo: TodoEntity): Promise<Required<TodoEntity> | Error> {
+        try{
```

Comment:
This needs to be refactored:

Remove try catch and text sanity check and move sanity checks to TODO CRUD